source/postprocess/topography_to_file.cc

```diff
@@ -0,0 +1,290 @@
+        {
+          MPI_Send (&output_file.str()[0], output_file.str().size()+1, MPI_CHAR, 0, mpi_tag,
+                    this->get_mpi_communicator());
+        }
+
+      // if output_interval is positive, then update the last supposed output
+      // time
+      if (output_interval > 0)
+        {
+          // We need to find the last time output was supposed to be written.
+          // this is the last_output_time plus the largest positive multiple
+          // of output_intervals that passed since then. We need to handle the
+          // edge case where last_output_time+output_interval==current_time,
+          // we did an output and std::floor sadly rounds to zero. This is done
+          // by forcing std::floor to round 1.0-eps to 1.0.
+          const double magic = 1.0+2.0*std::numeric_limits<double>::epsilon();
+          last_output_time = last_output_time + std::floor((this->get_time()-last_output_time)/output_interval*magic) * output_interval/magic;
+        }
+
+      return std::pair<std::string, std::string> ("Topography min/max:",
+                                                  output_stats.str());
+    }
+
+    template <int dim>
+    void
+    TopographyToFile<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Topography");
+        {
+          prm.declare_entry ("Output to file", "false",
+                             Patterns::List(Patterns::Bool()),
+                             "Whether or not to write topography to a text file named named"
+                             "'topography.NNNNN' in the output directory");
+
+          prm.declare_entry ("Output statistics", "true",
+                             Patterns::List(Patterns::Bool()),
+                             "Whether or not to output topography statistics to screen and "
+                             "statistics file in the output directory");
+          prm.declare_entry ("Time between text output", "0.",
+                             Patterns::Double (0),
+                             "The time interval between each generation of "
+                             "text output files. A value of zero indicates "
+                             "that output should be generated in each time step. "
+                             "Units: years if the "
+                             "'Use years in output instead of seconds' parameter is set; "
+                             "seconds otherwise.");
+        }
+        prm.leave_subsection();
+      }
+      prm.leave_subsection();
+    }
+
+
+
+    template <int dim>
+    void
+    TopographyToFile<dim>::parse_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Postprocess");
+      {
+        prm.enter_subsection("Topography");
+        {
+          write_to_file        = prm.get_bool ("Output to file");
+          write_statistics       = prm.get_bool ("Output statistics");
```

Comment:
I think this parameter is currently not used. Either implement it, or remove the parameter.